common/src/main/java/org/opensearch/ml/common/connector/ConnectorAction.java

```diff
@@ -232,7 +234,7 @@ public static String getRemoteServerFromURL(String url) {
         return SUPPORTED_REMOTE_SERVERS_FOR_DEFAULT_ACTION_TYPES.stream().filter(url::contains).findFirst().orElse("");
     }
 
-    private void validateProcessFunctions(String remoteServer, String processFunction, String funcNameForWarnText) {
+    private void validateProcessFunctions(String endPointUrl, String remoteServer, String processFunction, String funcNameForWarnText) {
         if (isInBuiltProcessFunction(processFunction)) {
```

Comment:
Based on the [issue](https://github.com/opensearch-project/ml-commons/issues/2986), I believe we should block the connector creation if the input process function doesn't exist. But based on your PRs I didn't find such validation, am I missing something?

Reply:
I understand it's an optional field but the point is: if it shows up it's either a painless script or a build-in process function, not any other value. So I believe the validation should be try to compile the value to painless script, if exception occurred, then check if it's a valid value in Pre/PostProcessFunction. Without this validation, user still can pass any value to the process_function field and not error message letting them know it's not a correct value, during runtime they found the function not working at all not knowing the root cause.

Reply:
@zane-neo I agree with that, that's the perfect validation here. But there is a technical limitation to implement that.
As per my code walkthrough ScriptService instance is not available during the connector creation code flow. It will be available during the actual processing of  pre or post process functions(inside the class DefaultPreProcessFunction). Are there anyway I can get ScriptService instance in ConnectorActions class ?

Reply:
@zane-neo Any suggestion here, please ?